File: java/pot.java
import java.util.Scanner;

public class pot {
	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		int N = scan.nextInt();
		int X = 0;
		while (N-- > 0) {
			int P = scan.nextInt();
			int p = P % 10;
			P /= 10;
			X += Math.pow(P, p);
		}
		System.out.println(X);
		scan.close();
	}
}

File: ts/src/pot.ts
import readline = require('readline')

function formatText (num: number): void {
  const base = Math.floor(num / 10)
  const pow = num % 10
  sum += base ** pow
}

let lineCount = 0
let sum = 0

readline
  .createInterface({
    input: process.stdin,
    output: process.stdout
  })
  .on('line', (line: string) => {
    if (lineCount > 0) {
      formatText(parseInt(line))
    }
    lineCount++
  })
  .on('close', () => {
    console.log(sum)
    process.exit(0)
  })
